js/modules.js

```diff
@@ -15,52 +15,79 @@ CORE.create_module("email-nav", function (sb) {
 CORE.create_module("email-catalog", function (sb) {
     var emails;
 
+    var openEmail = function(e) {
+        sb.hide();
+        sb.notify({
+            type : "open-email",
+            data : e.currentTarget.getAttribute("data-email-id")
```

Comment:
I was thinking if the emails are stored server side, each would have a unique ID, not unique ID per user. I haven't really looked into your query method but based off what your saying, I'm guessing it uses a user id and an email id to fetch an email? If thats happening, then it adds some safety. But if it ONLY uses an email id to fetch, then thats a security hole